### PR TITLE
JAMES-2558 fasten spam assassin tests

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: third-party/spamassassin/src/test/java/org/apache/james/spamassassin/SpamAssassinExtension.java [hunk 1]
```diff
@@ -119,6 +129,7 @@ private void train(String user, Path folder, TrainingKind trainingKind) throws I
                 paths
                     .filter(Files::isRegularFile)
                     .map(Path::toFile)
+                    .parallel()
```
Comment (mbaechler) [inline third-party/spamassassin/src/test/java/org/apache/james/spamassassin/SpamAssassinExtension.java]: `parallel` for IO is not recommended

File: third-party/spamassassin/src/test/java/org/apache/james/spamassassin/SpamAssassinExtension.java [hunk 2]
```diff
@@ -146,6 +157,10 @@ public void sync(String user) throws UnsupportedOperationException, IOException,
         public void dump(String user) throws UnsupportedOperationException, IOException, InterruptedException {
             spamAssassinContainer.execInContainer("sa-learn", "--dump", "magic", "-u", user);
         }
+
+        public void dumpAll() throws UnsupportedOperationException, IOException, InterruptedException {
```
Comment (mbaechler) [inline third-party/spamassassin/src/test/java/org/apache/james/spamassassin/SpamAssassinExtension.java]: what dumpAll means ? I guess it's wiping data from spamassassin but that would be better to express it in less specific words

**Comment by chibenwa** [comment]
```
[INFO] apache-james-spamassassin .......................... SUCCESS [01:05 min]
```

instead of

```
[INFO] apache-james-spamassassin .......................... SUCCESS [02:06 min]
```

**Comment by chibenwa** [comment]
test this please

**Comment by trantienduchn** [comment]
```
[da0ef60f28ef3da9dd9be69de96fd552498a122d] [ERROR] Failures: 
[da0ef60f28ef3da9dd9be69de96fd552498a122d] [ERROR]   ManageableMailQueueContract.removeByRecipientShouldRemoveSpecificEmailWhenMultipleRecipients:560 
[da0ef60f28ef3da9dd9be69de96fd552498a122d] Expecting:
[da0ef60f28ef3da9dd9be69de96fd552498a122d]   <[]>
[da0ef60f28ef3da9dd9be69de96fd552498a122d] to contain exactly (and in same order):
[da0ef60f28ef3da9dd9be69de96fd552498a122d]   <["name2"]>
[da0ef60f28ef3da9dd9be69de96fd552498a122d] but could not find the following elements:
[da0ef60f28ef3da9dd9be69de96fd552498a122d]   <["name2"]>
```
test this please

**Comment by chibenwa** [comment]
Was :green_apple: 

test this please

**Comment by chibenwa** [comment]
```
[da0ef60f28ef3da9dd9be69de96fd552498a122d] [ERROR]   SetMessagesMethodTest.lambda$setMessagesShouldCreateMessageInOutboxWhenSendingMessage$3:2337 » ConcurrentModification
```

test this please

**Comment by trantienduchn** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
Can be considered :green_apple:

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**